### PR TITLE
{AKS} Hotfix: `az aks create`: Fix issue that `assign-kubelet-identity` option can’t work

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

Fix issue that — `assign-kubelet-identity` option can’t work in az aks create command

Fix code bug in custom kubelet identity introduced by PR #18615.
This code bug was caused by the dependent SDK being migrated to track 2 recently. We corrected this error by replacing a wrongly used class name and filling in a missing parameter.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by zhoxing-ms — COMMENTED** [review]
Could we describe the issue more specifically in the title or description of PR? Such as what content should be written in release note to describe this change?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -2008,9 +2008,10 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
     ManagedClusterIdentity = cmd.get_models('ManagedClusterIdentity',
                                             resource_type=ResourceType.MGMT_CONTAINERSERVICE,
                                             operation_group='managed_clusters')
-    ManagedClusterPropertiesIdentityProfileValue = cmd.get_models('ManagedClusterPropertiesIdentityProfileValue',
-                                                                  resource_type=ResourceType.MGMT_CONTAINERSERVICE,
-                                                                  operation_group='managed_clusters')
+    ComponentsQit0EtSchemasManagedclusterpropertiesPropertiesIdentityprofileAdditionalproperties = cmd.get_models(
+        'ComponentsQit0EtSchemasManagedclusterpropertiesPropertiesIdentityprofileAdditionalproperties',
+        resource_type=ResourceType.MGMT_CONTAINERSERVICE,
+        operation_group='managed_clusters')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2014]: I'd like to ask why we need to change this model?

**Comment by zhoxing-ms** [comment]
Could we add some tests for this issue case?

**Comment by FumingZhang** [comment]
> Could we describe the issue more specifically in the title or description of PR? Such as what content should be written in release note to describe this change?

The title and PR description have been updated.

**Comment by FumingZhang** [comment]
> Could we add some tests for this issue case?

We already have a [test case](https://github.com/FumingZhang/azure-cli/blob/fuming/release-fix-custom-kubelet/src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py#L4491) for this command. 

Unfortunately, the previous test was executed locally, and the command was overridden by aks-preview accidentally. Since there is no problem with the implementation in aks-preview, this code bug was not found. 

We are looking forward to setting up a live test check-in pipeline for azure-cli (like the one ([AZCLI AKS LIVE TEST V2](https://dev.azure.com/msazure/CloudNativeCompute/_build?definitionId=219428&_a=summary)) already set in aks-preview) to avoid such mistakes in the future.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**